src/main/java/io/openliberty/tools/intellij/actions/LibertyGeneralAction.java

```diff
@@ -65,13 +65,16 @@ public void actionPerformed(@NotNull AnActionEvent e) {
                 return;
             }
         }
-        if (buildFile == null) {
+
+        boolean isActionFromShiftShift = "GoToAction".equalsIgnoreCase(e.getPlace());
+
+        if (isActionFromShiftShift || buildFile == null) {
             buildFile = (VirtualFile) e.getDataContext().getData(Constants.LIBERTY_BUILD_FILE);
-            // if still null, prompt for user to select
-            if (buildFile == null) {
+            // if still null, and it is not from shift-shift, then prompt for user to select
+            if (isActionFromShiftShift || buildFile == null) {
```

Comment:
In what context other than shift+shift would it be valid for a user to be given a dialog to choose from a list of projects? I think there was a case where there's only one project and this code just happens to work if no build file is available from the context. If there's more than one project, the user would be asked to select from a list, regardless of what they selected in the tool window.

Reply:
I think that the commented code I included is misleading... it should be something like `if still null, or it is from shift-shift, then prompt for the user to select`. This verification is covering two scenarios... where there is no build file in the context yet, and then it can be set, or it is already a build file provided but we can't rely on that because it could be just selected from the Liberty panel, but we are on a shift-shift action. Resuming, this code is shared between the shift-shift and the common right-click on "Start dev mode" in Liberty panel, so the verification would cover both cases.